src/Uno.UWP/ApplicationModel/Chat/ChatMessageStore.Android.cs

```diff
@@ -0,0 +1,295 @@
+#if __ANDROID_11__
+				case "Body":
+					return Android.Provider.Telephony.ITextBasedSmsColumns.Body;
+				case "Type":
+					return Android.Provider.Telephony.ITextBasedSmsColumns.Type;
+				case "Read":
+					return Android.Provider.Telephony.ITextBasedSmsColumns.Read;
+				case "Seen":
+					return Android.Provider.Telephony.ITextBasedSmsColumns.Seen;
+				case "Status":
+					return Android.Provider.Telephony.ITextBasedSmsColumns.Status;
+				case "Address":
+					return Android.Provider.Telephony.ITextBasedSmsColumns.Address;
+				case "Date":
+					return Android.Provider.Telephony.ITextBasedSmsColumns.Date;
+				case "DateSent":
+					return Android.Provider.Telephony.ITextBasedSmsColumns.DateSent;
+#else
+				case "Body":
+					return Android.Provider.Telephony.TextBasedSmsColumns.Body;
+				case "Type":
+					return Android.Provider.Telephony.TextBasedSmsColumns.Type;
+				case "Read":
+					return Android.Provider.Telephony.TextBasedSmsColumns.Read;
+				case "Seen":
+					return Android.Provider.Telephony.TextBasedSmsColumns.Seen;
+				case "Status":
+					return Android.Provider.Telephony.TextBasedSmsColumns.Status;
+				case "Address":
+					return Android.Provider.Telephony.TextBasedSmsColumns.Address;
+				case "Date":
+					return Android.Provider.Telephony.TextBasedSmsColumns.Date;
+				case "DateSent":
+					return Android.Provider.Telephony.TextBasedSmsColumns.DateSent;
+#endif
+				default:
+					throw new ArgumentException("TextBasedSmsColumns called with unknown column name");
+			}
+		}
+
+#region Changing default SMS app
+
+		internal const int RequestCode = 7001;
+		private static string _previousDefaultApp = "";
+		private static TaskCompletionSource<Android.Content.Intent?>? _currentSwitchSMSappRequest;
+
+		private string GetCurrentDefaultSMSapp()
+		{
+			var context = Android.App.Application.Context;
+			string? appName =  Android.Provider.Telephony.Sms.GetDefaultSmsPackage(context);
+			if(appName is null)
+			{
+				return "";
+			}
+			return appName;
+		}
+
+		/// <summary>
+		///  Switch default SMS app to newAppName (as in context.PackageName)
+		///  Return false if switching is unsuccesful.
+		///  To be "candidate" for default SMS app, your app must define several services/activities.
+		///  See https://stackoverflow.com/questions/21720657/how-to-set-my-sms-app-default-in-android-kitkat
+		/// </summary>
+		public async Task<bool> SwitchDefaultSMSapp(string newAppName)
+		{
+			if (!(Uno.UI.ContextHelper.Current is Android.App.Activity appActivity))
+			{
+				throw new InvalidOperationException("Application activity is not yet set, SwitchDefaultSMSapp called too early.");
+			}
+
+			if(GetCurrentDefaultSMSapp() == newAppName)
+			{
+				return true;
+			}
+
+			Android.Content.Intent intent;
+
+			if (Android.OS.Build.VERSION.SdkInt < Android.OS.BuildVersionCodes.Kitkat)
+			{
+				throw new NotImplementedException("OS.Build SDK is too low (I can work with  API > 18, Android Kitkat or newer).");

```

Comment:
```suggestion
				throw new NotSupportedException("Android SDK is too low (SwitchDefaultSMSapp only supports API level 18 or later, Android Kitkat or newer).");
```

Reply:
But it is not about Android SDK (used in build), it is about Android operating system version?

Reply:
Can I "resolve conversation"?

Reply:
Indeed, my apologies. The rest of the change is still valid.

Reply:
Other changes you requested are already implemented.